cmd/integration/commands/stages.go

```diff
@@ -367,7 +368,8 @@ func newSync(quitCh <-chan struct{}, db ethdb.Database, hook stagedsync.ChangeSe
 }
 
 func newBlockChain(db ethdb.Database) (*params.ChainConfig, *core.BlockChain, error) {
-	blockchain, err1 := core.NewBlockChain(db, nil, params.MainnetChainConfig, ethash.NewFaker(), vm.Config{}, nil, nil, nil)
+	txCacher := core.NewTxSenderCacher(runtime.NumCPU())
```

Comment:
We should not need `txCacher` here, because we have stage3 for recovery. Can you make it optional and not create it here?